### PR TITLE
strftime: Added support for the %w format specifier.

### PR DESCRIPTION
**Pull request by fjpanag**
## Summary

Enhanced `strftime()` with support for the `%w` format specifier.

## Impact

New functionality.

## Testing

Tested prints on simulator.

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by acassis** [comment]
@fjpanag : man strftime:
%w     The  day  of  the  week  as  a  decimal, range 0 to 6, Sunday being 0. 

Should be 0 to 6, not 00 to 06, please fix!

**Comment by acassis** [comment]
Tested on Linux:
```
#include <stdio.h>
#include <time.h>

int main () {
   time_t rawtime;
   struct tm *info;
   char buffer[80];

   time( &rawtime );

   info = localtime( &rawtime );

   strftime(buffer,80,"%x - %I:%M%p wday=%w", info);
   printf("Formatted date & time : |%s|\n", buffer );

   return(0);
}
```

File: libs/libc/time/lib_strftime.c [hunk 1]
```diff
@@ -398,6 +399,14 @@ size_t strftime(FAR char *s, size_t max, FAR const char *format,
              }
              break;
 
+           /* %w: The weekday as a decimal number (range 00 to 06). */
+
+           case 'w':
+             {
+               len = snprintf(dest, chleft, "%02d", tm->tm_wday);
```
Comment (acassis) [inline libs/libc/time/lib_strftime.c:406]:
```suggestion
               len = snprintf(dest, chleft, "%01d", tm->tm_wday);
```

**Review by acassis — CHANGES_REQUESTED** [review]

**Comment by fjpanag** [comment]
>

**Comment by fjpanag** [comment]
> @fjpanag : man strftime: `%w The day of the week as a decimal, range 0 to 6, Sunday being 0.`
> 
> Should be 0 to 6, not 00 to 06, please fix!

I was pretty sure that I saw it formatted with two-digits, but apparently your are correct.  
Fixed!

**Comment by fjpanag** [comment]
> @fjpanag : man strftime: `%w The day of the week as a decimal, range 0 to 6, Sunday being 0.`
> 
> Should be 0 to 6, not 00 to 06, please fix!

I was pretty sure that I saw it formatted with two-digits, but apparently your are correct.  
Fixed!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**